tests/unit/test_charm.py

```diff
@@ -20,6 +20,8 @@ def setUp(self):
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
 
+    @patch("charm.NginxIngressCharm._delete_unused_ingresses")
```

Comment:
Thanks for sharing the blog post. Since would be out of the scope of this PR, I added the autospec parameter only to the mocks created by this PR. I believe we could open a new story for refactoring the tests, what do you think?

Reply:
Yeh it is probably reasonable to do so, although this is kind of digging a deeper hole. The use of mocks her looks excessive which could indicate that the unit tests should actually be integration tests or that we need to write a few abstractions for the external dependencies that are easier to mock. Any thoughts @gregory-schiano?

Reply:
Yeap perhaps doing like Weii did on Spring Boot Charm, creating some mock classes 
https://github.com/canonical/spring-boot-k8s-operator/blob/main/tests/unit/spring_boot_patch.py

Globally speaking if we start needing more mock classes like this, it could be helpful to make a "Charm Mock module" in a separate repo we could easily import and use instead of defining those in multiple charms

Reply:
ISD-417 was created to address this.